tools/make_class_power.py

```diff
@@ -0,0 +1,206 @@
+
+    #Make the power spectra module
+    engine = CLASS.ClassEngine(pre_params)
+    powspec = CLASS.Spectra(engine)
+    bg = CLASS.Background(engine)
+
+    #Save directory
+    sdir = os.path.split(paramfile)[0]
+    #Get and save the transfer functions if needed
+    if config['DifferentTransferFunctions'] == 1.:
+        trans = powspec.get_transfer(z=redshift)
+        tfile = os.path.join(sdir, config['FileWithTransferFunction'])
+        save_transfer(trans, tfile, bg, redshift)
+        transfut = powspec.get_transfer(z=config['InputFutureRedshift'])
+        tfile = os.path.join(sdir, config['FileWithFutureTransferFunction'])
+        save_transfer(transfut, tfile, bg, config['InputFutureRedshift'])
+    #Get and save the matter power spectrum
+    mink = 2*math.pi/boxmpc/100
+    kk = np.logspace(np.log10(mink), np.log10(maxk), max(config['Ngrid']*2,300))
+    pk_lin = powspec.get_pklin(k=kk, z=redshift)
+    pkfile = os.path.join(sdir, config['FileWithInputSpectrum'])
+    if os.path.exists(pkfile):
+        raise IOError("Refusing to write to existing file: ",pkfile)
+    np.savetxt(pkfile, np.vstack([kk, pk_lin]).T)
+
+def save_transfer(transfer, transferfile, bg, redshift):
+    """Save a transfer function. Note we save the CAMB FORMATTED transfer functions.
+    These can be generated from CLASS by passing the 'format = camb' on the command line.
+    The transfer functions differ by:
+        T_CAMB(k) = -T_CLASS(k)/k^2 """
+    if os.path.exists(transferfile):
+        raise IOError("Refusing to write to existing file: ",transferfile)
+    #This format matches the default output by CAMB and CLASS command lines.
+    #Some entries may be zero sometimes
+    kk = transfer['k']
+    ftrans = np.zeros((np.size(transfer['k']), 9))
+    ftrans[:,0] = kk
+    ftrans[:,1] = -1*transfer['d_cdm']/kk**2
+    ftrans[:,2] = -1*transfer['d_b']/kk**2
+    ftrans[:,3] = -1*transfer['d_g']/kk**2
+    ftrans[:,4] = -1*transfer['d_ur']/kk**2
+    #This will fail if there are no massive neutrinos present
+    try:
+        #We use the most massive neutrino species, since these
+        #are used for initialising the particle neutrinos.
+        ftrans[:,5] = -1*transfer['d_ncdm[2]']/kk**2
+    except ValueError:
+        pass
+    omegacdm = bg.Omega_cdm(redshift)
+    omegab = bg.Omega_b(redshift)
+    omeganu = bg.Omega_ncdm(redshift)
+    #Note that the CLASS total transfer function apparently includes radiation!
+    #We do not want this for the matter power: we want CDM + b + massive-neutrino.
+    ftrans[:,6] = -1*(omegacdm *transfer['d_cdm'] + omegab * transfer['d_b'] + omeganu * ftrans[:,5])/(omeganu + omegacdm + omegab)/kk**2
+    #The CDM+baryon weighted density.
+    ftrans[:,7] = -1*(omegacdm *transfer['d_cdm'] + omegab * transfer['d_b'])/(omegacdm + omegab)/kk**2
+    ftrans[:,8] = -1*transfer['d_tot']/kk**2
+
+    np.savetxt(transferfile, ftrans)
+
+if __name__ == "__main__":
+    extpk = None
```

Comment:
Could you add argparse and a main function -- it's actually no more code than this but easier to read.

Reply:
ok!